### ISSUE TITLE
Backlinks

### ISSUE THREAD

**Issue by HanEmile** [hedgedoc/hedgedoc#3266]
### Is your feature request related to a problem? Please describe.

I've got a lot of pads (1000+) which link to each other. Finding pads isn't much of an issue as they are linked, but being able to see what other pads link to the current one would be amazing.

### Describe the solution you'd like

It would be amazing to have [backlinks](https://en.wikipedia.org/wiki/Backlink) for pads. This would be a list of pads that link to a specific pad. The links could be added like this (kind of like the table-of-contents `[toc]` directive:

```
[backlinks]
```

which would render to the following:

```
- [title of pad 1](./pad1)
- [title of pad 2](./pad2)
- ...
```

A few important things to take into accout:
- permissions: don't leak information from pads that might not be public
- depth: a single level should be enough for the beginning

### Describe alternatives you've considered

I've considered building this as an external script which works on all pads externally, indexes them and updates the pads as needed.

I've also just exported all pads in my history and added them to external tools such as obsidian, but having this internally would be quite nice.

### Additional context

Further thought's would be options to access more information on the document itself, such as forward links (a collection of links to somewhere else).

**Comment by HanEmile** [comment]
Regarding implementing this: There are two main paths that could be followed:

- do some kind of full-text search on all pads as soon as you open a pad (tbh.: that's not as nice, as real-time updates aren't possible when working in multiple pads at once)
- have some metadata with a pad into which backlinks can be added by other pads (if a link to a pad A is created in some other pad B, the backlink can be added to A, this is nice, as this also works when a link is removed (and thus the backlink does not exist anymore)).

TL;DR:
- **real-time** updates would be nice
- **removing pads** should be considered

**Comment by davidmehren** [comment]
I don't think we are going to implement this. In addition to implementation complications, this feels like a feature for a wiki, which we are not.

**Comment by kaspar030** [comment]
> In addition to implementation complications, this feels like a feature for a wiki, which we are not.

If the implementation complications could be solved, would you reconsider? "backlinks" use the same information as e.g., a ['graph view](https://github.com/hedgedoc/hedgedoc/issues/3854) would (link relations between pads).

I'd been using hedgedoc as PKM tool for a while. Backlinks as feature were missing, though. I've since tried Logseq and Obsidian, but they're basically single-user and make it difficult to collaborate.
Just adding backlinks would bring Hedgedoc a huge step towards being a nice, collaborative knowledge management tool.

**Comment by HanEmile** [comment]
> Backlinks as feature were missing, though
Backlinks are implemented in hedgedoc (right panel per default).

Implementing backlinks themselves for sure isn't simple, a possible option could be to have some kind of "metadata" on pads that can be altered by editing other pads, so for example if you link to another pad B from pad A, you would alter the metadata of pad B inserting the information that there is a link from pad A there. (removing them is another task that has to be considered, as said, this ain't easy :D).

Even if this looks like a wiki feature, I'd still argue that it might be really practical for hedgedoc, maybe just not yet. It might be nice to consider clarifying if interactions between pads are even an option that is to be considered or if hedgedoc is supposed to stay "pad isolated".
